rules/windows/registry/registry_set/registry_set_suspicious_env_variables.yml

```diff
@@ -0,0 +1,44 @@
+title: Suspicious Environment Variable Has Been Registered
+id: 966315ef-c5e1-4767-ba25-fce9c8de3660
+status: test
+description: Detects the creation of user-specific or system-wide environement variables via the registry. Which contains suspicious commands and strings
+references:
+    - https://infosec.exchange/@sbousseaden/109542254124022664
+author: Nasreddine Bencherchali
+date: 2022/12/20
+tags:
+    - attack.defense_evasion
+    - attack.persistence
+logsource:
+    product: windows
+    category: registry_set
+detection:
+    selection_main:
+        EventType: SetValue
+        TargetObject|contains: '\Environment\'
+    selection_details:
+        Details:
+            - 'powershell'
+            - 'pwsh'
+        Details|contains:
+            # Add more suspicious strings in env variables below
+            - '\AppData\Local\Temp\'
+            - 'C:\Users\Public\'
+            # Base64 MZ Header
+            - 'TVqQAAMAAAAEAAAA' # MZ..........
+            - 'TVpQAAIAAAAEAA8A'
+            - 'TVqAAAEAAAAEABAA'
+            - 'TVoAAAAAAAAAAAAA'
+            - 'TVpTAQEAAAAEAAAA'
+            # Base64 Invoke- (UTF-8)
+            - 'SW52b2tlL'
+            - 'ludm9rZS'
+            - 'JbnZva2Ut'
+            # Base64 Invoke- (UTF-16LE)
+            - 'SQBuAHYAbwBrAGUALQ'
+            - 'kAbgB2AG8AawBlAC0A'
+            - 'JAG4AdgBvAGsAZQAtA'
```

Comment:
```suggestion
    selection_details:
        - Details:
            - 'powershell'
            - 'pwsh'
        - Details|contains:
            # Add more suspicious strings in env variables below
            - '\AppData\Local\Temp\'
            - 'C:\Users\Public\'
            # Base64 MZ Header
            - 'TVqQAAMAAAAEAAAA' # MZ..........
            - 'TVpQAAIAAAAEAA8A'
            - 'TVqAAAEAAAAEABAA'
            - 'TVoAAAAAAAAAAAAA'
            - 'TVpTAQEAAAAEAAAA'
            # Base64 Invoke- (UTF-8)
            - 'SW52b2tlL'
            - 'ludm9rZS'
            - 'JbnZva2Ut'
            # Base64 Invoke- (UTF-16LE)
            - 'SQBuAHYAbwBrAGUALQ'
            - 'kAbgB2AG8AawBlAC0A'
            - 'JAG4AdgBvAGsAZQAtA'
        - Details|startswith:  # https://gist.github.com/Neo23x0/6af876ee72b51676c82a2db8d2cd3639
            - 'SUVY'
            - 'SQBFAF'
            - 'SQBuAH'
            - 'cwBhA'
            - 'aWV4'
            - 'aQBlA'
            - 'R2V0'
            - 'dmFy'
            - 'dgBhA'
            - 'dXNpbm'
            - 'H4sIA'
            - 'Y21k'
            - 'cABhAH'
            - 'Qzpc'
            - 'Yzpc'
```

Reply:
Shouldn't the different Details be ORed? I mean, a TVqQAA in an ENV variable is suspicious enough. It doesn't require an additional powershell or pwsh string. In an env var these strings alone should be suspicious enough, I'd say.

Reply:
I've also added some strings from my cheat sheet. Adding them with "Details|startswith" should be specific enough.

Reply:
Yes, they should be ORed you're right. I missed the `-` for that.

Reply:
Also great additions. The `startswith` is the better modifier for cases like those